src/main/java/io/confluent/kafkarest/v2/AvroKafkaConsumerState.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.kafkarest.v2;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.Properties;
+
+import io.confluent.kafka.serializers.KafkaAvroDecoder;
+import io.confluent.kafkarest.converters.AvroConverter;
+import io.confluent.kafkarest.entities.AvroConsumerRecord;
+import io.confluent.kafkarest.ConsumerInstanceId;
+import io.confluent.kafkarest.KafkaRestConfig;
+import io.confluent.kafkarest.ConsumerRecordAndSize;
+
+import kafka.serializer.Decoder;
+import kafka.utils.VerifiableProperties;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/**
+ * Avro implementation of KafkaConsumerState, which decodes into GenericRecords or primitive types.
+ */
+public class AvroKafkaConsumerState extends KafkaConsumerState<Object, Object, JsonNode, JsonNode> {
+
+  // Note that this could be a static variable and shared, but that causes tests to break in
+  // subtle ways because it causes state to be shared across tests, but only for the consumer.
+  private Decoder<Object> decoder = null;
+
+  public AvroKafkaConsumerState(KafkaRestConfig config,
+                           ConsumerInstanceId instanceId,
+                           Consumer consumer) {
+    super(config, instanceId, consumer);
+    Properties props = new Properties();
+    props.setProperty("schema.registry.url",
+                      config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
+  }
+
+  @Override
+  protected Decoder<Object> getKeyDecoder() {
```

Comment:
Decoders aren't used for the new consumer, so we can get rid of these. The equivalent for the new consumer would be Deserializers.

Same for Binary and JSON versions of this class.